### DISCUSSION TITLE
How to setup github actions to run my python script on schedule?

### DISCUSSION THREAD

**Discussion by KeshariPiyush24** [community/community#26539] [Actions]
<p><strong>I have a Github repo which analyzes data of COVID-19 cases from an api which updates everyday.</strong></p>
<p><strong>I have run.py script which clones the data but I don’t know how to setup github actions to run the script daily(automaticaly) so that data gets updated daily.</strong></p>
<p><strong>And one more thing I also want to export my Analysis.ipnb as index.html daily at the same time.</strong></p>
<p><strong>I don’t have any idea how to setup github actions to achive above tasks</strong></p>
<p><a href="https://github.com/PiyushKe/COVID-19-Outbreak-India" rel="nofollow noopener">Github Repo</a> </p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/piyushke">@piyushke</a> ,</p>
<p>Thank you for being here! To answer your query:</p>
<ol>
<li>
<p>You need to create a workflow yaml, doc here: <a href="https://help.github.com/en/actions/configuring-and-managing-workflows/configuring-a-workflow#creating-a-workflow-file" rel="noopener nofollow ugc">https://help.github.com/en/actions/configuring-and-managing-workflows/configuring-a-workflow#creating-a-workflow-file</a></p>
</li>
<li>
<p>In the yaml, set ‘schedule’ event. Details <a href="https://help.github.com/en/actions/reference/events-that-trigger-workflows#scheduled-events-schedule" rel="noopener nofollow ugc">here</a>. code sample as below:</p>
</li>
</ol>
<pre><code class="lang-auto">    name: py
    
    on:
      schedule:
        - cron: "0 0 * * *" #runs at 00:00 UTC everyday
    
    jobs:
      build:
        runs-on: ubuntu-latest
    
        steps:
          - name: checkout repo content
            uses: actions/checkout@v2 # checkout the repository content to github runner.
          - name: setup python
            uses: actions/setup-python@v2
            with:
              python-version: 3.8 #install the python needed
          - name: execute py script # run the run.py to get the latest data
            run: |
              python run.py
            env:
              key: ${{ secrets.key }} # if run.py requires passwords..etc, set it as secrets
          - name: export index
            .... # use crosponding script or actions to help export.
</code></pre>
<p>Thanks.</p>

**Comment by JasonAwFL** [comment]
<p>Hi <a class="mention" href="/piyushke">@piyushke</a>,</p>
<p>May I know where is you store the run.py file? Is it under github/workflow folder?</p>

**Comment by andresmitre** [comment]
<p>you can write a path <code>./scripts/run.py</code> or either put it on the main folder.  For example:</p>
<pre><code class="lang-auto">run: |
./scripts/run.py
</code></pre>

**Reply by hkalya** [reply]
Hi, I'm facing similar issue. I'm trying  to run simple  python script. Which ever folder I place the script and call, I keep getting this error message
Though the same script is present in both the folders
python: can't open file './script/testpython.py': [Errno 2] No such file or directory
Error: Process completed with exit code 2.

Can you please help?
